### PR TITLE
add API data enrichment to diff command

### PR DESCRIPTION
**Pull request by ajwootto**
It looks like this:

DevCycle Variable Changes:

⚠️   1 Variable With Notices
✅  2 Variables Added
❌  2 Variables Removed
🧹  1 Variable Cleaned up

⚠️  Notices

  1. Variable "no-exists" does not exist on DevCycle

✅ Added

  1. exists
	   Type: String
	   Location: test/utils/diff/sampleDiff.js:L1
  2. no-exists ⚠️
	   Location: test/utils/diff/sampleDiff.js:L2

❌ Removed

  1. exists2
	   Type: String
	   Location: test/utils/diff/sampleDiff.js:L1
  2. no-exists2 🧹
	   Location: test/utils/diff/sampleDiff.js:L2

🧹 Cleaned Up

The following variables that do not exist in DevCycle were cleaned up:

  1. no-exists2

### CODE REVIEW

**Comment by laurawarr** [comment]
It's not immediately obvious what the "cleaned up" list means

**Comment by ajwootto** [comment]
> It's not immediately obvious what the "cleaned up" list means

should i add a single sentence explainer to that section as well? Maybe at that point I could just add one to all sections?

**Comment by laurawarr** [comment]
> > It's not immediately obvious what the "cleaned up" list means
> 
> should i add a single sentence explainer to that section as well? Maybe at that point I could just add one to all sections?

What if all variables are contained in added/removed, but Notices calls out any potential issues.
Something like:

DevCycle Variable Changes:
⚠️ 2 Variables With Notices
✅ 2 Variables Added
❌ 2 Variables Removed

⚠️ Notices
1. Variable no-exists does not exist in DevCycle
2. Unable to identify the variable associated with VARIABLE.RANDOM

✅ Added
1. exists
  Type: String
  Location: test/utils/diff/sampleDiff.js:L1
2. no-exists ⚠️
  Location: test/utils/diff/sampleDiff.js:L2
  
❌ Removed
1. exists2
  Type: String
  Location: test/utils/diff/sampleDiff.js:L1
2. VARIABLE.RANDOM ⚠️
  Location: test/utils/diff/sampleDiff.js:L2

**Comment by ajwootto** [comment]
> > > It's not immediately obvious what the "cleaned up" list means
> > 
> > 
> > should i add a single sentence explainer to that section as well? Maybe at that point I could just add one to all sections?
> 
> What if all variables are contained in added/removed, but Notices calls out any potential issues. Something like:
> 
> DevCycle Variable Changes: ⚠️ 2 Variables With Notices ✅ 2 Variables Added ❌ 2 Variables Removed
> 
> ⚠️ Notices
> 
> 1. Variable no-exists does not exist in DevCycle
> 2. Unable to identify the variable associated with VARIABLE.RANDOM
> 
> ✅ Added
> 
> 1. exists
>    Type: String
>    Location: test/utils/diff/sampleDiff.js:L1
> 2. no-exists ⚠️
>    Location: test/utils/diff/sampleDiff.js:L2
> 
> ❌ Removed
> 
> 1. exists2
>    Type: String
>    Location: test/utils/diff/sampleDiff.js:L1
> 2. VARIABLE.RANDOM ⚠️
>    Location: test/utils/diff/sampleDiff.js:L2

that could work, would we not have a cleanup section in that case?

**Comment by laurawarr** [comment]
> > > > It's not immediately obvious what the "cleaned up" list means
> > > 
> > > 
> > > should i add a single sentence explainer to that section as well? Maybe at that point I could just add one to all sections?
> > 
> > 
> > What if all variables are contained in added/removed, but Notices calls out any potential issues. Something like:
> > DevCycle Variable Changes: ⚠️ 2 Variables With Notices ✅ 2 Variables Added ❌ 2 Variables Removed
> > ⚠️ Notices
> > 
> > 1. Variable no-exists does not exist in DevCycle
> > 2. Unable to identify the variable associated with VARIABLE.RANDOM
> > 
> > ✅ Added
> > 
> > 1. exists
> >    Type: String
> >    Location: test/utils/diff/sampleDiff.js:L1
> > 2. no-exists ⚠️
> >    Location: test/utils/diff/sampleDiff.js:L2
> > 
> > ❌ Removed
> > 
> > 1. exists2
> >    Type: String
> >    Location: test/utils/diff/sampleDiff.js:L1
> > 2. VARIABLE.RANDOM ⚠️
> >    Location: test/utils/diff/sampleDiff.js:L2
> 
> that could work, would we not have a cleanup section in that case?

I don't think we need a separate section for it — what do you think? We could use a different icon rather than a warning in that case, but I also wouldn't want to over complicate the comment

File: test/commands/diff.test.ts [hunk 1]
```diff
@@ -1,12 +1,16 @@
...
+import { AUTH_URL, BASE_URL } from '../../src/api/common'
+
+process.env = {}
 
 const expected = `
 DevCycle Variable Changes:
 
-✅ 4 Variables Added
-❌ 1 Variable Removed
+⚠️   0 Variables With Notices
```
Comment (laurawarr) [inline test/commands/diff.test.ts:10]: In the case that we don't have notices, I think we should omit this line

Comment (ajwootto) [reply test/commands/diff.test.ts:10]: agreed

**Review by laurawarr — APPROVED** [review]

**Pull request merged**